client/my-sites/customer-home/components/full-screen-launchpad.tsx

```diff
@@ -1,44 +1,50 @@
-import { CircularProgressBar } from '@automattic/components';
+import { CircularProgressBar, ConfettiAnimation } from '@automattic/components';
+import { updateLaunchpadSettings } from '@automattic/data-stores';
 import { Launchpad } from '@automattic/launchpad';
 import { Button } from '@wordpress/components';
 import { useI18n } from '@wordpress/react-i18n';
 import { useSelector } from 'react-redux';
-import { useGetDomainsQuery } from 'calypso/data/domains/use-get-domains-query';
 import useHomeLayoutQuery from 'calypso/data/home/use-home-layout-query';
 import { skipLaunchpad } from 'calypso/landing/stepper/utils/skip-launchpad';
 import { useDispatch } from 'calypso/state';
 import { requestSite } from 'calypso/state/sites/actions';
 import { getSite } from 'calypso/state/sites/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import { AppState } from 'calypso/types';
-import { useCelebrateLaunchModal } from '../cards/launchpad/use-celebrate-launch-modal';
 import { useLaunchpad } from '../cards/launchpad/use-launchpad';
-import CelebrateLaunchModal from './celebrate-launch-modal';
-
 import './full-screen-launchpad.scss';
 
-export const FullScreenLaunchpad = ( { onClose }: { onClose: () => void } ) => {
+export const FullScreenLaunchpad = ( { onClose }: { onClose: () => void } ): JSX.Element | null => {
 	const dispatch = useDispatch();
 	const { __ } = useI18n();
 	const siteId = useSelector( getSelectedSiteId ) || 0;
 	const site = useSelector( ( state: AppState ) => getSite( state, siteId ) );
 	const checklistSlug = site?.options?.site_intent ?? '';
+	const layout = useHomeLayoutQuery( siteId || null );
 
 	const launchpadContext = 'customer-home';
 
-	const { siteSlug, isDismissed, numberOfSteps, completedSteps, launchpadTitle } = useLaunchpad( {
+	const {
+		siteSlug,
+		isDismissed,
+		numberOfSteps,
+		completedSteps,
+		launchpadTitle,
+		hasChecklist,
+		refetch,
+	} = useLaunchpad( {
 		checklistSlug,
 		launchpadContext,
 	} );
 
-	const layout = useHomeLayoutQuery( siteId || null );
-	const { isOpen, setModalIsOpen, handleSiteLaunched } = useCelebrateLaunchModal( siteId, layout );
-	const { data: allDomains = [] } = useGetDomainsQuery( site?.ID ?? null, {
-		retry: false,
-	} );
+	const onSiteLaunched = async () => {
+		await updateLaunchpadSettings( siteId, {
+			checklist_statuses: { site_launched: true },
+		} );
 
-	const onSiteLaunched = () => {
-		handleSiteLaunched( !! site?.is_wpcom_atomic );
+		await refetch?.();
```

Comment:
This change to refetch the tasks might be removed soon since there is talk of removing the `Launch site` task